index.ts

```diff
@@ -192,34 +193,45 @@ interface Card {
   name: string;
 }
 
-let rooms: Room[] = [];
+let joinRooms: {[roomId: string]: Room} = {};
+let viewRooms: {[roomId: string]: Room} = {};
 wss.on("request", async function (request) {
+  const roomId = request.resourceURL.path.substr(1);
+  const readOnly = !!viewRooms[roomId];
   let name: string, id: string;
-  try {
-    let expressReq = request.httpRequest as express.Request;
-    ({name, id} = await new Promise((resolve, reject) => {
-      sessionParser(expressReq, {} as express.Response, () => {
-        if (expressReq.session && expressReq.session.name && expressReq.session.uid) {
-          resolve({name: expressReq.session.name as string, id: expressReq.session.uid as string});
-        } else {
-          reject("session, name or id not set");
-        }
-      })
-    }));
-  } catch (e) {
-    console.error("could not retrieve user name or id from session! Rejecting request...", request);
-    request.reject(403);
-    return;
+  if (!readOnly) {
+    try {
+      let expressReq = request.httpRequest as express.Request;
+      ({name, id} = await new Promise((resolve, reject) => {
+        sessionParser(expressReq, {} as express.Response, () => {
+          if (expressReq.session && expressReq.session.name && expressReq.session.uid) {
+            resolve({name: expressReq.session.name as string, id: expressReq.session.uid as string});
+          } else {
+            reject("session, name or id not set");
+          }
+        })
+      }));
+    } catch (e) {
+      console.error("could not retrieve user name or id from session! Rejecting request...", request);
+      request.reject(403);
+      return;
+    }
   }
   let connection = request.accept("stimmung", request.origin);
-  const roomId = request.resourceURL.path.substr(1);
-  if (!rooms[roomId]) {
-    rooms[roomId] = {
-      connections: [],
-      cards: [],
-    };
+  let room: Room;
+  if (readOnly) {
+    room = viewRooms[roomId];
+  } else {
+    if (!joinRooms[roomId]) {
+      let newRoom = {
+        connections: [],
+        cards: [],
+      };
+      let viewId = crypto.createHash('sha256').update(roomId).digest('hex');
+      viewRooms[viewId] = (joinRooms[roomId] = newRoom);
```

Comment:
no need for this construct

```suggestion
      viewRooms[viewId] = newRoom;
      joinRooms[roomId] = newRoom;